app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt

```diff
@@ -0,0 +1,271 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.material.icons.filled.Menu
+import androidx.compose.material.icons.filled.Warning
+import androidx.compose.material3.BottomSheetScaffold
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.rememberBottomSheetScaffoldState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.lifecycle.viewmodel.compose.viewModel
+import ch.hikemate.app.R
+import ch.hikemate.app.model.route.ListOfHikeRoutesViewModel
+import org.osmdroid.config.Configuration
+import org.osmdroid.events.MapListener
+import org.osmdroid.events.ScrollEvent
+import org.osmdroid.events.ZoomEvent
+import org.osmdroid.util.GeoPoint
+import org.osmdroid.views.MapView
+
+object MapScreen {
+  const val TEST_TAG_MAP = "map"
+  const val TEST_TAG_MENU_BUTTON = "menuButton"
+  const val TEST_TAG_HIKES_LIST = "hikesList"
+  const val TEST_TAG_HIKE_ITEM = "hikeItem"
+}
+
+@Composable
+fun MapScreen(
+    hikingRoutesViewModel: ListOfHikeRoutesViewModel =
+        viewModel(factory = ListOfHikeRoutesViewModel.Factory)
+) {
+  val context = LocalContext.current
+  // Avoid re-creating the MapView on every recomposition
+  val mapView = remember { MapView(context) }
+
+  Configuration.getInstance().apply {
+    // Set user-agent to avoid rejected requests
+    userAgentValue = context.packageName
+
+    // Allow for faster loading of tiles. Default OSMDroid value is 2.
+    tileDownloadThreads = 4
+
+    // Maximum number of tiles that can be downloaded at once. Default is 40.
+    tileDownloadMaxQueueSize = 40
+
+    // Maximum number of bytes that can be used by the tile file system cache. Default is 600MB.
+    tileFileSystemCacheMaxBytes = 600L * 1024L * 1024L
+  }
+
+  mapView.apply {
+    controller.setZoom(15.0)
+    // TODO : Set the center to the user's location
```

Comment:
Nitpick: Perhaps remove the "todo" and replace it with a marker, which you can reference from a task which you create in the product backlog.

Reply:
I'm not sure about what you mean in "replace it with a marker", what is a marker exactly? Should I leave a comment but simply remove the TODO and use another keyword?

Reply:
I think it would be cleaner to only have everything written once in the backlog item. If we have two places where the task is mentioned, we won't think about updating both in case we do update the task.

I could simply mention the line or lines to change directly in the issue or the backlog item, such as :

```kotlin
mapView.apply {

    // Other config options

    // TODO : Update this line to center the map on the user's location
    controller.setCenter(GeoPoint(46.5, 6.6))
    
    // Other config options
  }
```

Reply:
I created [this draft task](https://github.com/orgs/HikeMate/projects/1/views/2?pane=issue&itemId=83415053) in the backlog, let me know what you think

Reply:
This looks great, thanks!